tephra-core/src/main/java/org/apache/tephra/persist/AbstractTransactionLog.java

```diff
@@ -165,26 +203,36 @@ private void sync() throws IOException {
       // prevent writer being dereferenced
       tmpWriter = writer;
 
-      List<Entry> currentPending = getPendingWrites();
-      if (!currentPending.isEmpty()) {
-        tmpWriter.commitMarker(currentPending.size());
-      }
-
-      // write out all accumulated entries to log.
-      for (Entry e : currentPending) {
-        tmpWriter.append(e);
-        entryCount++;
-        latestSeq = Math.max(latestSeq, e.getKey().get());
+      Entry[] currentPending = getPendingWrites();
+      if (currentPending != null) {
+        entryCount = currentPending.length;
+        startTimerIfNeeded(tmpWriter, entryCount);
+        tmpWriter.commitMarker(entryCount);
+        for (Entry e : currentPending) {
+          tmpWriter.append(e);
+          latestSeq = Math.max(latestSeq, e.getKey().get());
```

Comment:
Is the `Math.max` actually needed? The ordering of the sequence number is the always increasing and is maintained in the queue (hence the array here). I think we just need to have `latestSeq = currentPending[currentPending.length - 1].getKey().get()` after the loop, right?

Reply:
yes, actually that has always been the case, even before these changes...